docs/public/specifications/test-zones/Address-TP/address03.md

```diff
@@ -0,0 +1,206 @@
+  * ns4.child’s IP addresses have no PTR records; the reverse zone is
+    configured to provoke NODATA responses on PTR queries by making the
+    expected node an empty non-terminal.
+  * ns5.child’s IP addresses have no PTR records; the reverse zone is
+    configured to provoke NXDOMAIN responses on PTR queries.
+
+### INCOMPLETE-PTR-1
+
+One of the name servers has PTR records for its IPv6 address, but not for its IPv4 address.
+
+* Zone: incomplete-ptr-1.address03.xa
+  * Delegated to: ns1.child and ns6.child.
+  * ns6.child’s IPv4 address has no PTR record, but its IPv6 address does.
+
+### INCOMPLETE-PTR-2
+
+One of the name servers has PTR records for its IPv4 address, but not for its IPv6 address.
+
+* Zone: incomplete-ptr-2.address03.xa
+  * Delegated to: ns1.child and ns7.child.
+  * ns7.child’s IPv4 address has a PTR record, but its IPv6 address does not.
+
+### NON-MATCHING-NAMES
+
+One of the name servers has PTR records pointing to a different hostname.
+
+* Zone: non-matching-names.address03.xa
+  * Delegated to: ns8.child and ns9.child.
+  * ns8.child’s IP addresses have a single PTR record, but its hostname is
+    different from the name server’s name.
+  * ns9.child’s IP addresses have more than one PTR record, and each of them
+    has a hostname different from the name server’s name.
+
+### PTR-IS-GOOD-CNAME-1
+
+One of the name servers has PTR records pointing to an alias whose target is in the same reverse zone.
+
+* Zone: ptr-is-good-cname-1.address03.xa
+  * Delegated to: ns1.child and ns10.child.
+  * ns10.child’s IP addresses have reverse entries that are aliased to another
+    node in the same zone. In other words, resolving the PTR resource records
+    for their IP addresses returns a CNAME resource record and the PTR record
+    after walking the CNAME chain.
+
+### PTR-IS-GOOD-CNAME-2
+
+One of the name servers has PTR records pointing to an alias whose target lives in a different zone.
+
+* Zone: ptr-is-good-cname-2.address03.xa
+  * Delegated to: ns1.child and ns11.child.
+  * ns11.child’s IP addresses have reverse entries that are aliased to another
+    node in a different zone. In other words, resolving the PTR resource
+    records for their IP addresses returns only a CNAME resource record,
+    therefore Zonemaster needs to do another query for the name at the target
+    of the CNAME resource record.
+
+### PTR-IS-DANGLING-CNAME
+
+One of the name servers has PTR records pointing to an alias whose target does not exist.
+
+* Zone: ptr-is-dangling-cname.address03.xa
+  * Delegated to: ns5.child and ns12.child.
+  * ns5.child is configured as described in the NO-NS-HAVE-PTR scenario.
+  * ns12.child’s IP addresses have reverse entries that are aliased to a
+    nonexistent node. In other words, there is a CNAME pointing to a node
+    that does not exist.
+
+### PTR-IS-ILLEGAL-CNAME
+
+One of the name servers has PTR records pointing to an alias with more than just one CNAME resource record.

```

Comment:
```suggestion
The reverse name of one of the name servers' IP address has multiple CNAME
records.
```

Reply:
Yes, very poor wording on my part. But the suggestion is incorrect: both the IPv4 and IPv6 addresses of the affected name servers have reverse names containing 2 CNAME RRs.